### PR TITLE
[RFC] Fixes for possible issues found with clang-check

### PR DESCRIPTION
**Pull request by michal-narajowski**
These are proposals for fixes for some issues found with clang-check. They may not be necessary. Comments are welcome.

### CODE REVIEW

**Comment by spoonofpower** [comment]
test this please

**Comment by spoonofpower** [comment]
test this please

File: net/nimble/controller/src/ble_ll_ctrl.c [hunk 1]
```diff
@@ -630,12 +630,20 @@ ble_ll_ctrl_phy_update_ind_make(struct ble_ll_conn_sm *connsm, uint8_t *dptr,
         CONN_F_PHY_UPDATE_SCHED(connsm) = 1;
 
         /* Set new phys to use when instant occurs */
-        connsm->phy_data.new_tx_phy = m_to_s;
-        connsm->phy_data.new_rx_phy = s_to_m;
-
         /* Convert m_to_s and s_to_m to masks */
-        m_to_s = 1 << (m_to_s - 1);
-        s_to_m = 1 << (s_to_m - 1);
+        if (m_to_s != 0) {
```
Comment (sjanc) [inline net/nimble/controller/src/ble_ll_ctrl.c]: I'd leave assignment unconditional and only make conversions under
connsm->phy_data.new_tx_phy = m_to_s;
connsm->phy_data.new_rx_phy = s_to_m;

if (m_to_s) {
  m_to_s = 1 << (m_to_s - 1);
}

if (s_to_m) {
    s_to_m = 1 << (s_to_m - 1);
}

**Comment by spoonofpower** [comment]
test this please

**Comment by michal-narajowski** [comment]
Thanks, Szymon. Updated.

**Pull request merged**